language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ContextTypeResolver.java

```diff
@@ -643,6 +643,11 @@ public ContextTypeResolver(PositionedOperationContext context) {
 
     @Override
     public Optional<TypeSymbol> transform(MappingConstructorExpressionNode mappingConstructorExpressionNode) {
+        Optional<TypeSymbol> typeSymbol = context.currentSemanticModel()
+                .flatMap(semanticModel -> semanticModel.typeOf(mappingConstructorExpressionNode));
+        if (typeSymbol.isPresent() && typeSymbol.get().typeKind() != TypeDescKind.COMPILATION_ERROR) {
+            return typeSymbol;
+        }
```

Comment:
We can rewrite this part as below:
```
return context.currentSemanticModel()
                .flatMap(semanticModel -> semanticModel.typeOf(mappingConstructorExpressionNode))
                .filter(typeSymbol -> typeSymbol.typeKind() != TypeDescKind.COMPILATION_ERROR)
                .or(() -> mappingConstructorExpressionNode.parent().apply(this));
```
Btw, I have done the same change in #34958. Depending on which PR gets merged first, let's rebase.

Reply:
If rebasing, itll leave the PR only with a test case. In that case, do we still need the PR?

Reply:
We need the test case for sure :smile: So, we need the PR.